src/core/utils/hooks/useContainerWidth.tsx

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useMergedRefs } from './useMergedRefs';
+import { useResizeObserver } from './useResizeObserver';
+
+/**
+ * Hook that returns the width of an element in three stages:
+ *  - initialized with 0
+ *  - immediately set to element's initial width as soon as it's mounted
+ *  - update to new width every time it changes (using `useResizeObserver` hook)
+ *
+ * @private
+ * @param watchResizes If false, ResizeObserver will not be connected and only the initial width will be returned
+ * @returns [ref to attach to the element, stateful width of the element]
+ *
+ * @example
+ * const [ref, width] = useContainerWidth();
+ * ... // do something with width
+ * return <div ref={ref}>...</div>;
+ */
+export const useContainerWidth = <T extends HTMLElement>(
```

Comment:
Not sure about the naming here. Maybe should be `useResponsiveWidth` or `useContentWidth` or `useResizableWidth`.

Also not sure if we should include height too. Usually for responsive design, width is the main concern.

Reply:
hmm.. yeah, I think this hook is more about giving you the width of the element on screen change, but then naturally i have a question, why not just use resize observer for that.. so then maybe would vote for `useResizableWidth`, but.. kinda sounds weird.

Reply:
I had `useResizableWidth` previously but then I thought it could be useful in non-resizable cases too so I changed it